app/scripts/presenters/visit_presenter.js

```diff
@@ -0,0 +1,102 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+define([
+  'underscore'
+], function (_) {
+  'use strict';
+
+  var MINIMUM_IMAGE_ENTROPY = 1.75;
+  var MINIMUM_IMAGE_WIDTH = 270;
+  var EXTRACTED_IMAGE_DENY_PATTERN = /github.com|(\..{2,3}\/$)|([&?]q=)|([&?]page=\d+)/;
+
+  function VisitPresenter (visit) {
```

Comment:
This whole file may go away at some point. Most of these decisions/features will end up implemented on the server side.